qt-models/filtermodels.cpp

```diff
@@ -369,6 +372,7 @@ void LocationFilterModel::addName(const QString &newName)
 	list.prepend(newName);
 	setStringList(list);
 	checkState.insert(checkState.begin(), true);
+	emit dataChanged(createIndex(0, 0), createIndex(0, 0));
```

Comment:
Probably moving the line `checkState.insert(checkState.begin(), true);` before `setStringList()` is enough?

I'm not happy about this function - it generates some temporary state that will be invalidated immediately afterwards. There must be a more sensible way...

Reply:
and again